minidump-processor/src/processor.rs

```diff
@@ -479,5 +647,94 @@ where
         unknown_streams,
         unimplemented_streams,
         symbol_stats,
-    })
+    };
+
+    // Report the unwalked result
+    if let Some(reporter) = options.stat_reporter {
+        reporter.add_unwalked_result(&state);
+    }
+
+    {
+        let memory_list = &memory_list;
+        let modules = &state.modules;
+        let system_info = &state.system_info;
+        let unloaded_modules = &state.unloaded_modules;
+        let options = &options;
+
+        futures_util::future::join_all(
+            state
+                .threads
+                .iter_mut()
```

Comment:
wow, I’m even surprised the borrowchecker lets you do this, but okay ;-)

Reply:
yes this is the Super Cool Thing about the rust iterator API, each yielded reference has its lifetime tied to the original collection, and not the iterator, so

```
let a = iter_mut.next();
let b = iter_mut.next();
*a = 1;
*b = 2;
```

Is totally valid!

This is only sound because collection iterators are "one-shot" and even if you mix `next` vs `next_back`, you'll only ever get one reference to each element, and need to discard them all before getting a new iterator. It's basically a generalization of `slice::split_at_mut`. :)